identity-iota/src/client/client.rs

```diff
@@ -160,16 +160,22 @@ impl Client {
 
     let auth: AuthChain = AuthChain::try_from_messages(did, &messages)?;
 
-    let diff: DiffChain = if auth.current().immutable() {
-      DiffChain::new()
-    } else {
+    // Check if there is any query given and return
+    let diff_option = match did.query() {
```

Comment:
You can use [`query_pairs`](https://docs.rs/did_url/0.1.0/did_url/struct.DID.html#method.query_pairs) for this - it's pretty explicit but it handles encoding automatically

```rust
let skip_diff: bool = did.query_pairs().any(|(key, _)| key == "diff" && value == "false").unwrap_or_default();
```

Reply:
Ah thank you! I didn't see query_pairs. But is has to be:
`let skip_diff: bool = did.query_pairs().any(|(key, value)| key == "diff" && value == "false");`
Is that right?

Reply:
Yeah, that looks right 👍